manager/controlapi/node.go

```diff
@@ -273,7 +273,8 @@ func (s *Server) RemoveNode(ctx context.Context, request *api.RemoveNodeRequest)
 		if node == nil {
 			return grpc.Errorf(codes.NotFound, "node %s not found", request.NodeID)
 		}
-		if node.Spec.Role == api.NodeRoleManager {
+		member := s.raft.GetMemberByNodeID(request.NodeID)
```

Comment:
Not sure I understand the suggestion exactly, but if you see a more logical way feel free to make the change.